README.md

```diff
@@ -31,5 +31,5 @@ The [project homepage](http://janusgraph.org) contains more information on Janus
 
 ## Contributing
 
-Please see [`CONTRIBUTING.md`](CONTRIBUTING.md) for more information, including
```

Comment:
Why are we renaming `CONTRIBUTING.md` to `GETTING_INVOLVED.md`?

If there's a file called `CONTRIBUTING` or `CONTRIBUTING.md` at the root of the repo, GitHub handles it [specially by auto-linking it](https://github.com/blog/1184-contributing-guidelines).

Reply:
https://github.com/blog/2111-issue-and-pull-request-templates specifically notes that you can keep the PR and issue template files in `.github` subdir to keep the root of the repo leaner by having fewer files there.

I would like to propose:

* `CONTRIBUTING.md` stays at the root of the repo because it's generally useful to everyone
* PR and issue template files go into `.github` because they're GitHub-specific workflow items

@amp, @pluradj — thoughts?